### PR TITLE
support BQCC spark

### PR DESCRIPTION
**Pull request by 600lyy**
### Change description
support BQCC spark

### Tests you have done
KCC_USE_DIRECT_RECONCILERS=BigQueryConnectionConnection hack/compare-mock fixture/sparkconnectionbasic

- [x] Run `make ready-pr` to ensure this PR is ready for review.
- [x] Perform necessary E2E testing for changed resources.

### CODE REVIEW

**Comment by 600lyy** [comment]
@yuwenma pls help review

File: apis/refs/v1beta1/metastoreserviceref.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		key.Namespace = obj.GetNamespace()
+	}
+
+	// NOTYET: Currently there is no MetaStoreService KCC resource
+	service := &unstructured.Unstructured{}
+	service.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "metastore.cnrm.cloud.google.com",
+		Version: "v1beta1",
+		Kind:    "MetastoreService",
+	})
```
Comment (yuwenma) [inline apis/refs/v1beta1/metastoreserviceref.go:88]: I don't see a KCC reference doc for this resource. Do you mind sharing the document you found?

File: pkg/controller/direct/bigqueryconnection/connection_controller.go [hunk 2]
```diff
@@ -105,6 +105,32 @@ func (m *model) AdapterForObject(ctx context.Context, reader client.Reader, u *u
...
+				if err != nil {
+					return nil, err
+				}
+				obj.Spec.SparkSpec.SparkHistoryServer.DataprocClusterRef.External = cluster.String()
+			}
+		}
+	}
+
+	// Resolve Spark.MetastoreServiceRef
+	if obj.Spec.SparkSpec != nil {
```
Comment (yuwenma) [inline pkg/controller/direct/bigqueryconnection/connection_controller.go:122]: nit: suggest merge this if with line109

File: apis/refs/v1beta1/metastoreserviceref.go (hunk 1 above)
Comment (600lyy) [reply apis/refs/v1beta1/metastoreserviceref.go:88]: There is no KCC resource today, shall I remove it?

Comment (yuwenma) [reply apis/refs/v1beta1/metastoreserviceref.go:88]: > There is no KCC resource today,
If that's the case, you can only include `external` field but not name and namespace. See [the 1st bullet point](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/docs/develop-resources/api-conventions/resource-reference.md#rule-2-required-fields)

Do you know which value the REST API accept?

Comment (600lyy) [reply apis/refs/v1beta1/metastoreserviceref.go:88]: The REST API accpets the format of: projects/{projectNumber}/locations/{locationId}/services
does it look like a typical one?

https://cloud.google.com/dataproc-metastore/docs/reference/rest/v1beta/projects.locations.services/create

Comment (yuwenma) [reply apis/refs/v1beta1/metastoreserviceref.go:88]: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/docs/develop-resources/api-conventions/resource-reference.md#basic-rules 
In such case, we still want to use resource reference but only with the `External` field. This gives us the option to extend the field to support KCC object once the resource is there.

**Review by yuwenma — COMMENTED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**